src/zenml/artifact_stores/local_artifact_store.py

```diff
@@ -12,30 +26,145 @@
 
 from zenml.artifact_stores import BaseArtifactStore
-from zenml.enums import ArtifactStoreFlavor
-from zenml.io import fileio
+from zenml.artifact_stores.base_artifact_store import (
+    ArtifactStoreInterfaceError,
+)
+
+PathType = Union[bytes, str]
 
 
 class LocalArtifactStore(BaseArtifactStore):
     """Artifact Store for local artifacts."""
 
-    supports_local_execution = True
-    supports_remote_execution = False
+    # Class Configuration
+    FLAVOR: ClassVar[str] = "local"
+    SUPPORTED_SCHEMES: ClassVar[Set[str]] = {""}
+
+    @staticmethod
+    def open(name: PathType, mode: str = "r") -> Any:
+        """Open a file at the given path."""
+        return open(name, mode=mode)
+
+    @staticmethod
+    def copyfile(src: PathType, dst: PathType, overwrite: bool = False) -> None:
+        """Copy a file from the source to the destination."""
+        if not overwrite and os.path.exists(dst):
+            raise OSError(
+                (
+                    "Destination file %r already exists and argument "
+                    "`overwrite` is false."
+                )
+                % dst
+            )
+        shutil.copyfile(src, dst)  # type: ignore[type-var, arg-type]
+
+    @staticmethod
+    def exists(path: PathType) -> bool:
+        """Returns `True` if the given path exists."""
+        return os.path.exists(path)
+
+    @staticmethod
+    def glob(pattern: PathType) -> List[PathType]:
+        """Return the paths that match a glob pattern."""
+        return glob.glob(pattern)  # type: ignore[type-var]
+
+    @staticmethod
+    def isdir(path: PathType) -> bool:
+        """Returns whether the given path points to a directory."""
+        return os.path.isdir(path)
+
+    @staticmethod
+    def listdir(path: PathType) -> List[PathType]:
+        """Returns a list of files under a given directory in the filesystem."""
+        return os.listdir(path)  # type:ignore[return-value]
+
+    @staticmethod
+    def makedirs(path: PathType) -> None:
+        """Make a directory at the given path, recursively creating parents."""
+        os.makedirs(path, exist_ok=True)
+
+    @staticmethod
+    def mkdir(path: PathType) -> None:
+        """Make a directory at the given path; parent directory must exist."""
+        os.mkdir(path)
+
+    @staticmethod
+    def remove(path: PathType) -> None:
+        """Remove the file at the given path. Dangerous operation."""
+        os.remove(path)
+
+    @staticmethod
+    def rename(src: PathType, dst: PathType, overwrite: bool = False) -> None:
+        """Rename source file to destination file.
+        Args:
+            src: The path of the file to rename.
+            dst: The path to rename the source file to.
+            overwrite: If a file already exists at the destination, this
+                method will overwrite it if overwrite=`True`
+        """
+        if not overwrite and os.path.exists(dst):
+            raise OSError(
+                (
```

Comment:
Maybe an f-string instead? ALso not sure if an OSError is the best here or if it should be a `FileExistsError` instead?

Reply:
Converted it to an f-string and a FileExistsError.